tests/test_surface/test_new_surface.py

```diff
@@ -0,0 +1,89 @@
+import pytest
+import numpy as np
+import hypothesis.strategies as st
+from hypothesis import given, assume
+from xtgeo.surface.regular_surface import RegularSurface
+
+
+def test_default_values():
+    surf = RegularSurface(2, 2, 0.0, 0.0)
+    assert surf.values.data.tolist() == [[0.0, 0.0], [0.0, 0.0]]
+
+
+@pytest.mark.parametrize(
+    "input_val, expected_result",
+    [
+        (1, [[1.0, 1.0], [1.0, 1.0]]),
+        ([1, 2, 3, 4], [[1.0, 2.0], [3.0, 4.0]]),
+        (np.zeros((2, 2)), [[0.0, 0.0], [0.0, 0.0]]),
+        (None, [[0.0, 0.0], [0.0, 0.0]]),
+        (
+            np.ma.MaskedArray([[2, 2], [2, 2]], mask=[[True, False], [False, True]]),
+            [[2.0, 2.0], [2.0, 2.0]],
+        ),
+    ],
+)
+def test_values_type(input_val, expected_result):
+    surf = RegularSurface(2, 2, 0.0, 0.0, values=input_val)
+    assert isinstance(surf.values, np.ma.MaskedArray)
```

Comment:
Do we also want to assert that the masked array is the same? (especially for the one that provides some other than all `False`)